src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -1992,8 +1992,8 @@ private static void SplitName(string? fullname, out string? name, out string? ns
                 return;
 
             // Get namespace
-            int nsDelimiter = fullname.LastIndexOf(".", StringComparison.Ordinal);
-            if (nsDelimiter != -1)
+            int nsDelimiter = fullname.LastIndexOf('.');
```

Comment:
Good catch! Do we already have an analyzer for similar cases? If not, should we add such?

Reply:
I think the closest we have is https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1847, for string.Contains.  I'm not sure why we didn't extend that to {Last}IndexOf. @jeffhandley, was this discussed?